### PR TITLE
doc: Virual minibatch size and multiple GPUs

### PR DESCRIPTION
**Pull request by boeddeker**
I made a proposal for documenting multiple GPUs.

What do you think? (I didn't check for typos)

### CODE REVIEW

**Comment by jensheit** [comment]
Perhaps you should introduce the name virtual minibatch earlier to clarify from the beginning that both uses of a minibatch are not equivalent. It is a little confusing that both use cases are called minibatch.

File: doc/virtual_batch_size_multi_gpu.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    if i == virtual_minibatch_size:                 # <-----
+        i = 0                                       # <-----
+        optimizer.step()                            # <-----
+        optimizer.zero_grad()
+```
+
+Both options for the minibatch size can be combined.
+The effective minibatch size for the optimizer will be the dataset minibatch size times the `virtual_minibatch_size`. For operations like batch normalization (i.e. operations that work in the minibatch axis) the batch size will be the minibatch that is used to produce the dataset, NOT the virtual batch size.
+
+## Why use (Virtual) minibatch size?
```
Comment (thequilo) [inline doc/virtual_batch_size_multi_gpu.md:48]: Mention that it is simliar to `accumulate_grads` (I think) from ESPnet?

Comment (thequilo) [reply doc/virtual_batch_size_multi_gpu.md:48]: It's called `accum_grad` in ESPnet

**Comment by jensheit** [comment]
> How do we handle proof-reading for grammar/typos/punctuation? I think it's not a good idea to add a comment for every typo.

Just directly commit the changes you want to do.

File: doc/virtual_batch_size_multi_gpu.md (hunk 1 above)
Comment (boeddeker) [reply doc/virtual_batch_size_multi_gpu.md:48]: Du you want to add this the comparison with ESPnet?

**Comment by boeddeker** [comment]
Can we merge?
